template/src/shared/components/responsiveThemeProvider/responsiveThemeProvider.component.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { FC, Fragment, useState } from 'react';
+import { DefaultTheme, ThemeProvider } from 'styled-components';
+import responsiveTheme from '../../../theme/responsiveTheme';
+import { WindowListener } from '../windowListener/windowListener.component';
+
+const parseTheme = (theme: object): DefaultTheme => responsiveTheme(theme);
+
+export interface ResponsiveThemeProviderProps {
+  theme: object;
+  children: React.ReactNode;
+}
+
+export const ResponsiveThemeProvider: FC<ResponsiveThemeProviderProps> = ({ theme: themeDefinition, children }) => {
+  const [theme, setTheme] = useState(parseTheme(themeDefinition));
+  const handleResize = () => setTheme(parseTheme(themeDefinition));
+
+  return (
+    <Fragment>
+      <WindowListener eventType="resize" throttle={200} onEvent={handleResize} />
```

Comment:
Maybe it would be better now to rewrite `WindowListener` to hook, for example `useWindowListener`? It's just an idea, we can create a new issue for that :)

Reply:
Good point 👍  I'll make an issue and do this in a separate PR